framework/test/unit/node/consensus/certificate_generation/commit_pool.spec.ts

```diff
@@ -332,9 +333,112 @@ describe('CommitPool', () => {
 	describe('getAggregageCommit', () => {
 		it.todo('');
 	});
-	describe('_aggregateSingleCommits', () => {
-		it.todo('');
+
+	describe('aggregateSingleCommits', () => {
+		const height = 45678;
+		const blockHeader1 = testing.createFakeBlockHeader({ height });
+		const blockHeader2 = testing.createFakeBlockHeader({ height });
+		const validatorInfo1 = {
+			address: getRandomBytes(20),
+			blsPublicKey: getRandomBytes(48),
+			blsSecretKey: getRandomBytes(32),
+		};
+		const validatorInfo2 = {
+			address: getRandomBytes(20),
+			blsPublicKey: getRandomBytes(48),
+			blsSecretKey: getRandomBytes(32),
+		};
+		const certificate1 = computeCertificateFromBlockHeader(blockHeader1);
+		const certificate2 = computeCertificateFromBlockHeader(blockHeader2);
+		let expectedCommit: AggregateCommit;
```

Comment:
Minor preference but prefer `let` to come after `const` declarations for readability